addOns/replacer/src/test/java/org/zaproxy/zap/extension/replacer/ReplacerParamUnitTest.java

```diff
@@ -0,0 +1,182 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.replacer;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.configuration.HierarchicalConfiguration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.zaproxy.zap.extension.replacer.ReplacerParamRule.MatchType;
+import org.zaproxy.zap.utils.ZapXmlConfiguration;
+
+/** Unit test for {@link ReplacerParam}. */
+class ReplacerParamUnitTest {
+
+    private static final ReplacerParamRule REPORT_TO_RULE =
+            new ReplacerParamRule(
+                    ReplacerParam.REPORT_TO_DESC,
+                    MatchType.RESP_HEADER_STR,
+                    ReplacerParam.REPORT_TO_REGEX,
+                    true,
+                    ReplacerParam.REPORT_TO_REPLACEMENT,
+                    null,
+                    false);
+    private ReplacerParam param;
+    private ZapXmlConfiguration configuration;
+
+    @BeforeEach
+    void setUp() {
+        param = new ReplacerParam();
+        configuration = new ZapXmlConfiguration();
+    }
+
+    @Test
+    void shouldHaveConfigVersionKey() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.getConfigVersionKey(), is(equalTo("replacer[@version]")));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"true", "false"})
+    void shouldLoadConfirmRemoveFromConfig(boolean state) {
+        // Given
+        configuration.setProperty("replacer.confirmRemoveToken", state);
+        // When
+        param.load(configuration);
+        // Then
+        assertThat(param.isConfirmRemoveToken(), is(equalTo(state)));
+    }
+
+    @Test
+    void shouldDefaultConfirmRemoveTrue() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.isConfirmRemoveToken(), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldHaveExpectedVersionInDefaultConfig() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.getCurrentVersion(), is(equalTo(1)));
+    }
+
+    @Test
+    void shouldHaveExpectedNumberOfRulesInDefaultConfig() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.getRules().size(), is(equalTo(4)));
+        ReplacerParamRule newRule = param.getRule(ReplacerParam.REPORT_TO_DESC);
+        assertNotNull(newRule);
+        assertThat(newRule.getDescription(), is(equalTo(ReplacerParam.REPORT_TO_DESC)));
+    }
+
+    @Test
+    void existingRuleShouldNotBeOverwritten() {
```

Comment:
`shouldNotOverwriteExistingRulesOnUpdate` and should have one from versioned as well, where it verifies pretty much the same (make it parameterized on the version).

Reply:
Is that case related to these changes?

You can set it on the config using the version attribute.

Reply:
I'm totally missing what you want here. Neither configuration nor param have a way to set version. Even if they did we're just starting v1 so how would I test v1 to v1.

Reply:
What's your vision? Add a -1 and remove the other test?

Reply:
null for no version and use the current numbers.

Reply:
Ok will tweak that. Otherwise good?

Reply:
Yes.

Reply:
Maybe I was not clear but the change was meant to merge both tests (I'd rename the parameterized to `shouldNotOverwriteExistingRulesOnUpdate`though).